src/compaction_filter.rs

```diff
@@ -15,11 +16,18 @@ pub trait CompactionFilter {
     /// indicates that this key-value should be removed from the
     /// output of the compaction.  The application can inspect
     /// the existing value of the key and make decision based on it.
-    fn filter(&mut self, level: usize, key: &[u8], value: &[u8]) -> bool;
+    fn filter(
+        &mut self,
+        level: usize,
+        key: &[u8],
+        value: &[u8],
+        new_value: &mut Vec<u8>,
+        value_changed: &mut bool,
+    ) -> bool;
 }
 
 #[repr(C)]
-pub struct CompactionFilterProxy {
+struct CompactionFilterProxy {
```

Comment:
Why not use Box<dyn CompactionFilter > directly?

Reply:
We need to associate a name to it.